docker-compose.yaml

```diff
@@ -103,6 +118,29 @@ services:
       - MINIO_ACCESS_KEY=minio
       - MINIO_SECRET_KEY=minio123
       - SPARK_MODE=notebook
-      - MAX_EXECUTORS=2
+      - MAX_EXECUTORS=4
+      - POSTGRES_USER=hive
+      - POSTGRES_PASSWORD=hivepassword
+      - POSTGRES_DB=hive
+      - POSTGRES_URL=postgres:5432
+    volumes:
+      - ./cdr/cdm/jupyter:/cdm_shared_workspace
+
+  postgres:
+    image: postgres:16.3
+    restart: always
+    container_name: postgres
+    # To avoid incorrect user permissions, manually create the volume directory before running Docker.
+    # export UID=$(id -u)
+    # export GID=$(id -g)
+    # mkdir -p cdr/cdm/jupyter/cdm-postgres
+    # reference: https://forums.docker.com/t/systemd-coredump-taking-ownership-of-tmp-db-directory-and-contents-in-rails-app/93609
+    user: "${UID}:${GID}"
+    ports:
+      - "5432:5432"
+    environment:
+      - POSTGRES_USER=hive
+      - POSTGRES_PASSWORD=hivepassword
+      - POSTGRES_DB=hive
     volumes:
-      - ./cdr/cdm/jupyter:/cdm_shared_workspace
\ No newline at end of file
+      - ./cdr/cdm/jupyter/cdm-postgres:/var/lib/postgresql/data
```

Comment:
Whty does the postgres data need to be shared?

Reply:
it doesn't need to. But I think it's convenient if we want to config postgresql on the fly if needed. Like what we did previously to manually delete lock files. Once system is settle, I am planning to mount it differently on rancher.

Reply:
I am also planning to build a user_notebook and a dev_notebook and mount differently so that users won't accidentally corrupt files.

Reply:
ok, maybe add a comment about that? If we're running postgres separately in rancher its data folders probably shouldn't be accessible in the notebook

Reply:
👍

Reply:
: